porcupine/plugins/langserver.py

```diff
@@ -412,6 +416,27 @@ def _handle_lsp_event(self, lsp_event: lsp.Event) -> None:
 
             for tab in self.tabs_opened.keys():
                 self._send_tab_opened_message(tab)
+
+            # TODO: this is a terrible hack:
+            #   - This only works for pyls, and kinda defeats the purpose of langservers.
+            #   - It should be configured in filetypes.toml.
+            #   - This causes an error elsewhere because sansio-lsp-client doesn't
+            #     officially support workspace/didChangeConfiguration yet.
+            #   - This doesn't refresh as venv changes.
+            self._lsp_client._send_request(
+                "workspace/didChangeConfiguration",
+                {
+                    "settings": {
+                        "pyls": {
+                            "plugins": {
+                                "jedi": {
+                                    "environment": str(python_venv.get_venv(self._id.project_root))
```

Comment:
- This can set `"environment": "None"`
- Should be made one step less hacky by moving to filetypes.toml